in_toto/models/metadata.py

```diff
@@ -137,14 +139,15 @@ def type_(self):
     return self.signed.type_
 
 
-  def sign(self, key):
+  def sign(self, key=None, signer: Signer=None):
```

Comment:
I appreciate the effort to maintain backwards compatibility, but I'm more in favor of maintaining a clean API.

My preferred solution would be to replace the current `sign(key: dict)` and `sign_gpg(keyid: str)` with a `sign(signer: Signer)` and maybe add a `isinstance(signer, Signer)` check with an error message that points out the breaking change. We can remove/reword the error message after a few releases.

If we don't want to break the API (yet), we can also add a new sign method with a different name and deprecate current `sign` and `sign_gpg` more slowly. 

In https://github.com/theupdateframework/python-tuf/pull/2010#issuecomment-1148363668 we had a similar problem. There we ended up with a mix of above two suggestion, i.e. create a new function with a different name and remove the old ones right away. The reason why we changed the name was that it was not as straight-forward as here to point out to the user that the function was called with the old argument types.

Reply:
cc @SantiagoTorres @adityasaky

Reply:
Maybe we don't need to worry too much about the signing interface of `Metablock` if we decide to henceforth only support creation of DSSE envelopes.